superset/viz.py

```diff
@@ -177,6 +177,11 @@ def process_metrics(self) -> None:
         self.all_metrics = list(self.metric_dict.values())
         self.metric_labels = list(self.metric_dict.keys())
 
+    def construct_where(self, query_obj: QueryObjectDict, predicate_string: str) -> str:
+        if query_obj:
+            predicate_string = "{} AND {}".format(query_obj, predicate_string)
```

Comment:
We probably need to add parenthesis here, if either of these has an OR it will break.
```suggestion
            predicate_string = "({}) AND ({})".format(query_obj, predicate_string)
```